### ISSUE TITLE
Deserialize of List of Guid

### ISSUE THREAD

**Issue by OlegNadymov** [esskar/Serialize.Linq#105]
It's impossible deserialize list of Guid.
Sample:
```
        [DataContract]
        private class SerializeTestObject
        {
            [DataMember] public Guid Identity { get; set; }

            [DataMember] public int Number { get; set; }

            [DataMember] public string Code { get; set; }
        }

        [TestMethod]
        public void LinqExpressionSerializeTest()
        {
            var listIds = new List<Guid> {Guid.NewGuid(), Guid.NewGuid()};
            Expression<Func<SerializeTestObject, bool>> e = x => listIds.Contains(x.Identity);

            var jsonSerializer = new Serialize.Linq.Serializers.JsonSerializer()
            {
                AutoAddKnownTypesAsListTypes = true
            };
            var expressionSerializer = new ExpressionSerializer(jsonSerializer, new FactorySettings {AllowPrivateFieldAccess = true});

            var expressionNodeText = expressionSerializer.SerializeText(e, new FactorySettings {AllowPrivateFieldAccess = true});

            var eAfter = expressionSerializer.DeserializeText(expressionNodeText);//<--- exception here
        }
```

Exception message. Text was translated from Russian by Google Translate:
```
The “8bd7501b-e106-41a6-a695-b18a869619a2” value type is not “System.Guid”, so it cannot be used in this basic collection.
Parameter name: value.
```

**Comment by OlegNadymov** [comment]
I've created PR for the issue: https://github.com/esskar/Serialize.Linq/pull/106

**Comment by esskar** [comment]
will be released with 1.7.3.1
thanks for your help
